include/mbedtls/aes.h

```diff
@@ -69,26 +69,50 @@ extern "C" {
 // Regular implementation
 //
 
+/** VIA padlock needs 16 extra bytes for alignment. */
+#if (defined(MBEDTLS_PADLOCK_C) && defined(MBEDTLS_HAVE_X86)) ||        \
+    (defined(MBEDTLS_AESNI_C) && MBEDTLS_AESNI_HAVE_CODE == 2)
```

Comment:
This actually doesn't work because `MBEDTLS_HAVE_X86` and `MBEDTLS_AESNI_HAVE_CODE` are defined in `padlock.h` and `aesni.h` respectively. So there is always no extra space when an alignment is needed.

I think the reason why only windows test failed on CI is related to the compiler. On Windows, we use MSVC which should support aesni intrinsics by default (not that sure). On Linux, we use GCC and CLANG which need extra options to enable the intrinsics implementation (e.g. `gcc -m32 -mpclmul -maes`). I think we lost the coverage of these case in `all.sh`, do we need to improve this? @tom-cosgrove-arm

Reply:
> On Linux, we use GCC and CLANG which need extra options to enable the intrinsics implementation (e.g. `gcc -m32 -mpclmul -maes`). I think we lost the coverage of these case in `all.sh`

Now the cases are covered by https://github.com/Mbed-TLS/mbedtls/pull/7630.